### PR TITLE
scripts/sign_encrypt.py: add flags for the encryption key type

### PR DESCRIPTION
**Pull request by hoiho-amzn**
Allow encryption key type to be overridden from command-line. Defaults
to SHDR_ENC_KEY_DEV_SPECIFIC.

Signed-off-by: Donald Chan <hoiho@amazon.com>

### CODE REVIEW

**Review by jforissier — COMMENTED** [review]
Hello @hoiho-amzn,
Thanks for the patch, a couple of comments below.

**Comment by hoiho-amzn** [comment]
All comments have been addressed

**Comment by jforissier** [comment]
Thanks for the update, LGTM. Please squash the two commits and add:
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

Note: the CI failure is unrelated to this PR, it can be safely ignored.

File: scripts/sign_encrypt.py [hunk 1]
```diff
@@ -43,7 +46,8 @@ def get_args(logger):
         ' TA image file.\n' +
         '                 Takes arguments --uuid, --ta-version, --in, --out,' +
         ' --key\n' +
-        '                 and --enc-key (optional).\n' +
+        '                 --enc-key (optional), and' +
```
Comment (b49020) [inline scripts/sign_encrypt.py:49]: Incorrect `,` placement.

File: scripts/sign_encrypt.py [hunk 2]
```diff
@@ -54,8 +58,8 @@ def get_args(logger):
         ' image file from\n' +
         '                 TA raw image and its signature. Takes' +
         ' arguments\n' +
-        '                 --uuid, --in, --key, --enc-key (optional), --out,' +
-        ' --algo (optional) and --sig.\n\n' +
+        '                 --uuid, --in, --key, --out, --enc-key (optional),' +
+        ' --enc-key-type (optional), --algo (optional) and --sig.\n\n' +
```
Comment (b49020) [inline scripts/sign_encrypt.py:62]: Above `digest` command requires similar changes as well.

**Review by b49020 — COMMENTED** [review]
With below nits fixed, feel free to add:

`Reviewed-by: Sumit Garg <sumit.garg@linaro.org>`

File: scripts/sign_encrypt.py (hunk 2 above)
Comment (hoiho-amzn) [reply scripts/sign_encrypt.py:62]: I didn't see where the incorrect `,` placement is for this one but feel free to point it out.

Comment (b49020) [reply scripts/sign_encrypt.py]: Actually I meant `digest` command requires `--enc-key-type (optional)` argument as well.

Comment (hoiho-amzn) [reply scripts/sign_encrypt.py]: done

**Pull request merged**